WultraMobileTokenSDK/Oidc/Utils/WMTOidcUtils.swift

```diff
@@ -0,0 +1,123 @@
+            URLQueryItem(name: "redirect_uri", value: config.redirectUri),
+            URLQueryItem(name: "scope", value: config.scopes),
+            URLQueryItem(name: "state", value: state),
+            URLQueryItem(name: "nonce", value: nonce),
+            URLQueryItem(name: "response_type", value: "code")
+        ]
+        
+        if let pkceCodes {
+            components.queryItems?.append(URLQueryItem(name: "code_challenge", value: pkceCodes.codeChallenge))
+            components.queryItems?.append(URLQueryItem(name: "code_challenge_method", value: pkceCodes.codeMethod))
+        }
+        
+        if let url = components.url {
+            D.debug("OIDC: Successfully created auth URL: \(url.absoluteString)")
+            return url
+        } else {
+            D.warning("OIDC: Failed to create URL.")
+            throw WMTError(reason: .authorizationUrlCreationFailed)
+        }
+    }
+    
+    /// Processes a deeplink URI to validate its state and extract OIDC activation attributes.
+    ///
+    /// - Parameters:
+    ///   - url: The deeplink URL received from the OIDC provider during the authorization process.
+    ///   - oidcAuthorizationData: Data containing the necessary data for the OIDC flow.
+    ///
+    /// - Returns: A `WMTOidcPowerAuthActivationAttributes` attributes needed for OIDC PowerAuth activation flow
+    /// - Throws: An error when attributes cannot be constructed.
+    public static func processWebCallback(from url: URL, with oidcAuthorizationData: WMTOidcAuthorizationRequest) throws -> WMTOidcPowerAuthActivationAttributes {
+        
+        guard let queryItems = URLComponents(string: url.absoluteString)?.queryItems else {
+            D.error("OIDC: Invalid callback URL: \(url)")
+            throw WMTError(reason: .invalidDeeplink)
+        }
+        
+        guard let code = queryItems.first(where: { $0.name == "code" })?.value else {
+            D.error("OIDC: Code not found in response from URL: \(url)")
+            throw WMTError(reason: .invalidDeeplink)
+        }
+        
+        guard let state = queryItems.first(where: { $0.name == "state" })?.value else {
+            D.error("OIDC: State not found in response from URL: \(url)")
+            throw WMTError(reason: .invalidDeeplink)
+        }
+        
+        guard state == oidcAuthorizationData.state else {
+            D.error("OIDC: Invalid 'state' in URL: \(url)")
+            throw WMTError(reason: .invalidDeeplink)
+        }
+        
+        return WMTOidcPowerAuthActivationAttributes(
+            providerId: oidcAuthorizationData.providerId,
+            code: code,
+            nonce: oidcAuthorizationData.nonce,
+            codeVerifier: oidcAuthorizationData.codeVerifier
+        )
+    }
+    
+    /// PKCE Helper
+    /// Helper: Generates a SHA-256-based code challenge.
+    private static func generateCodeChallenge(verifier: String) throws -> String {
```

Comment:
Only used once at highly specific place, his could be inlined in the function instead of standalone function.